pkgs/development/haskell-modules/configuration-common.nix

```
@@ -2657,7 +2657,7 @@ self: super: {
   tomland = doJailbreak super.tomland;
 
   llvm-ffi = super.llvm-ffi.override {
-    LLVM = pkgs.llvmPackages_13.libllvm;
+    LLVM = pkgs.llvmPackages_16.libllvm;
```

> You mean I should add notes to the llvm-ffi package as released to Hackage? Would it help to add a default.nix file to the llvm-ffi Cabal package?

Ah, I just mean to add an explanation to this nix file:

```suggestion
    # Make sure llvm-ffi uses LLVM-16, since LLVM-16 is both the default in Nixpkgs, and the latest version that llvm-ffi supports.
    LLVM = pkgs.llvmPackages_16.libllvm;
```

I tried to condense the reasoning from https://github.com/NixOS/nixpkgs/pull/227424, but I'm not quite sure I fully understood it, so feel free to fix this comment if it should be explained differently!

---

I suggest the following comment:

llvm-ffi has no way to autodetect location and preferred version of installed LLVM.
Any auto-detection makes fixing worse if it fails.
So we go the manual way.
If the default version of LLVM and the latest supported version of llvm-ffi match, then llvm-ffi just uses this one.
If the two versions differ, we have two options:

1. Choose the latest LLVM version that is supported by llvm-ffi. That is, users installing
llvm-ffi related packages risk to install two LLVM versions.

2. Use NixOS' default LLVM version and tell llvm-ffi so via a Cabal flag. E.g. if NixOS' default
LLVM version would be 15, then you must declare: `llvm-ffi = enableCabalFlag "llvm1500" super.llvm-ffi;`

For now we have chosen option 1.

---

@thielema can we add you as a maintainer for `llvm-ffi`? Then you'd be notified in the future if the package breaks.